sycl/doc/extensions/experimental/sycl_ext_matrix/sycl_ext_oneapi_matrix.asciidoc

```diff
@@ -0,0 +1,938 @@
+   joint_matrix<sub_group, int32_t, use::accumulator, tM, tN> tC;
+   joint_matrix_fill(sg, tC, 0);
+   for (int k = 0; k < K; k += tK) {
+     joint_matrix_load(sg, tA,
+          multi_ptr<int8_t, sycl::access::address_space::global_space>(memA) +
+          sg_startx * tM * K + k, K);
+     joint_matrix_load(sg, tB,
+          multi_ptr<int8_t, sycl::access::address_space::global_space>(memB) +
+          k * N + sg_starty/SG_SIZE*tN, N);
+     tC = joint_matrix_mad(sg, tA, tB, tC);
+   }
+   joint_matrix_apply(sg, tC, [=](int8_t x) {
+    x *= alpha;
+   });
+   joint_matrix_store(sg, tC,
+        multi_ptr<int32_t, sycl::access::address_space::global_space>(memC) +
+        sg_startx * tM * N + sg_starty/SG_SIZE*tN, N, layout::row_major);
+}).wait();
+```
+
+=== Query Interface
+Most devices support only certain values for the `Rows` and `Cols`
+template parameters and only certain types for the `T` template
+parameter. Moreover, most devices support only certain combinations of
+these template parameter for the A, B, C, and D matrices in the
+`joint_matrix_mad` function (see Appendix: Supported Combinations Per
+Hardware). This extension adds two query APIs that can be used to
+determine the set of legal parameters for a particular device. One
+form provides `constexpr` values for these parameters, which can be
+used when the application knows the specific device architecture on
+which it will run. The other form uses the standard information
+descriptor queries for the device object.
+
+The description below uses the terms `M`, `N`, and `K` to identify the
+matrix dimensions of a multiply and add operation `D = C + A*B`. The
+`D` and `C` matrices are `M` rows by `N` columns. The `A` matrix is
+`M` rows by `K` columns, and the `B` matrix is `K` rows by `N` columns.
+
+==== Compile-Time Query
+This returns `constexpr` values to use in `joint_matrix` template
+arguments but depends on an enumeration of the matrix hardware (See
+`sycl::ext::oneapi::experimental::architecture`) in the
+link:../sycl_ext_oneapi_device_architecture.asciidoc[sycl_ext_oneapi_device_architecture]
+extension that can be tested.
+The compile-time query interface proposed here consists of two
+functionalities:
+
+- Validation: at compile time, the validation functionality informs
+  the user whether a specific combination is valid or not. This takes
+  place when the user specifies all template parameters.
+
+- Default values: this provides a default shape if the user does not
+  provide a specific combination. In this case, aliases to the
+  `joint_matrix` type can be used, namely
+  `joint_matrix_a/b/c/d` where no additional argument is needed. This
+  form happens when the user specifies all template parameters except
+  the sizes of the matrices M, N, and K.
+
+The table below provides a description for each of the member
+variables in `matrix_params` class and the forms in which  they are
+defined.
+
+[frame="none",options="header"]
```

Comment:
I think commit a7f92ce2dc006914878f8dcc913afcb9a8a97213 was trying to fix the problem reported in [this comment](https://github.com/intel/llvm/pull/7964#discussion_r1282414732).  However, that commit changed the wrong table header.

This comment is pointing at the table header that needs to be changed.  The change should be something like:

```suggestion
[frame="none",options="header",cols="40%,60%"]
```

You'll have to play around with the percentages, though, to find something that looks good.  The goal is to avoid confusing line breaks in that first column.

Reply:
Fixed in ee282500c36070319828ff5d62e585f5fa5e887d